BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift

```diff
@@ -105,187 +151,256 @@ public class TransactionConfirmationStore: ObservableObject {
       reject(transaction: transaction, completion: { _ in })
     }
   }
-
-  func updateGasValue(for transaction: BraveWallet.TransactionInfo) {
-    let gasLimit = transaction.ethTxGasLimit
-    let gasFormatter = WeiFormatter(decimalFormatStyle: .gasFee(limit: gasLimit.removingHexPrefix, radix: .hex))
-    let gasFee: String
-    if transaction.isEIP1559Transaction {
-      gasFee = transaction.txDataUnion.ethTxData1559?.maxFeePerGas ?? ""
+  
+  @MainActor func prepare(with activeTransactionId: String) async {
```

Comment:
I don't think we need to pass in a `activeTransactionId` here. I think after fetching the transactions, we can just do a check if the `self.activeTransactionId` exists in `transactions` and if it doesn't we assign `firstTx.id`?